fenix/app/src/main/java/org/mozilla/fenix/translations/preferences/downloadlanguages/PrefDownloadLanguages.kt

```diff
@@ -0,0 +1,513 @@
+    val learnMoreText = stringResource(id = R.string.pref_download_languages_header_learn_more)
+    val learnMoreState = LinkTextState(
+        text = learnMoreText,
+        url = "www.mozilla.com",
+        onClick = {},
+    )
+    Box(
+        modifier = Modifier
+            .padding(
+                start = 72.dp,
+                top = 6.dp,
+                bottom = 6.dp,
+                end = 16.dp,
+            )
+            .semantics { heading() },
+    ) {
+        LinkText(
+            text = stringResource(
+                R.string.pref_download_languages_header,
+                learnMoreText,
+            ),
+            linkTextStates = listOf(learnMoreState),
+            style = FirefoxTheme.typography.subtitle1.copy(
+                color = FirefoxTheme.colors.textPrimary,
+            ),
+            linkTextDecoration = TextDecoration.Underline,
+        )
+    }
+}
+
+@Composable
+private fun prefDownloadLanguageItemContentDescription(
+    item: PrefDownloadLanguageItem,
+    itemDescription: String,
+): String {
+    val contentDescription: String
+
+    when (item.state.status) {
+        PrefDownloadLanguageItemStatus.Downloaded -> {
+            contentDescription =
+                item.languageModel.language?.localizedDisplayName + " " + itemDescription
+        }
+
+        PrefDownloadLanguageItemStatus.NotDownloaded -> {
+            contentDescription =
+                item.languageModel.language?.localizedDisplayName + " " + itemDescription
+        }
+
+        PrefDownloadLanguageItemStatus.DownloadInProgress -> {
+            contentDescription =
+                item.languageModel.language?.localizedDisplayName + " " + itemDescription + " " + stringResource(
+                    id = R.string.translations_bottom_sheet_translating_in_progress,
+                )
+        }
+
+        PrefDownloadLanguageItemStatus.Selected -> {
+            contentDescription =
+                item.languageModel.language?.localizedDisplayName + " " + itemDescription
+        }
+    }
+    return contentDescription
+}
+
+@Composable
+private fun PrefIconDownloadLanguageItem(item: PrefDownloadLanguageItem) {
+    when (item.state.status) {
+        PrefDownloadLanguageItemStatus.Downloaded -> {
+            if (item.state.type != PrefDownloadLanguageItemType.EnglishLanguage) {
+                Icon(
+                    painter = painterResource(
+                        id = R.drawable.ic_delete,
+                    ),
+                    contentDescription = null,
+                    tint = FirefoxTheme.colors.iconPrimary,
+                )
+            }
+        }
+
+        PrefDownloadLanguageItemStatus.NotDownloaded -> {
+            if (item.state.type != PrefDownloadLanguageItemType.EnglishLanguage) {
+                Icon(
+                    painter = painterResource(
+                        id = R.drawable.ic_download,
+                    ),
+                    contentDescription = null,
```

Comment:
Please update the `contentDescription` to "download" or else set one for the entire element.

TalkBack currently reads "[Language] [Size], button, double tap to activate", should be something like "[Language] [Size] download, button, double tap to activate".

Reply:
Tested with TalkBack, seems right to me now, thanks!